backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -329,8 +328,10 @@ public FutureExperimentRunDAO(
                 .execute());
   }
 
-  private InternalFuture<Void> checkProjectPermission(
-      List<String> projId, ModelDBActionEnum.ModelDBServiceActions action) {
+  private InternalFuture<Boolean> checkEntityPermissionBasedOnResourceTypes(
```

Comment:
```suggestion
  private InternalFuture<Boolean> getEntityPermissionBasedOnResourceTypes(
```
Now that you're returning something vs just checking.

Reply:
updated